healthcheck/dial.go

```diff
@@ -123,6 +142,13 @@ func dialTCP(network, addr string, timeout time.Duration, mark int) (nc net.Conn
 		}
 	}
 
+	// optionally set source address for tunneling mode healthcheck
+	if srcip != nil {
+		if err := syscall.Bind(c.fd, lsa); err != nil {
```

Comment:
this seems to me only setting the socket source address not the tunnel (outter) source address. Also because of the socket mark, I actually thinking the healthcheck is going through the DSR service created for this healthcheck.

Reply:
It's not for setting tunnel (outter) source address, but tunnel (inner) source address. So that after decapsulation by the backend, it knows which is the source IP address.

I think I wrongly used "outter" in updated comments above dialTCP(), will correct it back in the next update.